### ISSUE TITLE
vfb_nblast('VFB_00000001', target="GMR-Gal4", n = 10) not returning valid VFB ID's

### ISSUE THREAD

**Issue by Robbie1977** [jefferis/vfbr#10]
```
      id  score
1   <NA> 0.5375
2   <NA> 0.5124
3   <NA> 0.4930
4   <NA> 0.4894
5   <NA> 0.4648
6   <NA> 0.4431
7   <NA> 0.4423
8   <NA> 0.4393
9   <NA> 0.4230
10  <NA> 0.4226
```

**Comment by jefferis** [comment]
It doesn't work for GMR nblast. It does for flycircuit single neuron.

**Comment by jefferis** [comment]
@Robbie1977 I assume it should also return VFB ids for GMR lines?

**Comment by jefferis** [comment]
Let's assume I have `22A05`. What solr query should I use to get GMR ids @Robbie1977?

**Comment by Robbie1977** [comment]
should be the same as with flycircuit but with asterisks surrounding ```*22A05* ```

**Comment by Robbie1977** [comment]
wondering if it might be better to use neo4j for this as can specify exact types you want returning. This should be achieved in solr by filtering on id. 

The issue is avoiding genetic components being returned as well as the individual image

**Comment by Robbie1977** [comment]
@jefferis isn't this already returning the list: https://github.com/jefferis/vfbr/blob/master/R/vfbids.R#L71

**Comment by Robbie1977** [comment]
@jefferis you're just not using them (gmr_vfbid) in https://github.com/jefferis/vfbr/blob/master/R/vfb_nblast.R#L60

**Comment by jefferis** [comment]
Hmm, actually I currently use queries against `label_s` for flycircuit ids since no asterisks required.

**Comment by jefferis** [comment]
Take a look at 8a3b39bc5912b2de1158c71cc82bf2ae7aceeefb and see if you are happy.

**Comment by jefferis** [comment]
More generally is this VFB id always the one that I want to return for a GMR id? It seems tied to the particular image for that e.g. `VFB_00021621` =  `GMR_93D09_AE_01_03-fA01b`.

**Comment by Robbie1977** [comment]
Yep will work fine for now - a bit worried we might change labels to R10A09 format in longterm as Janelia have requested this.
Also as we add more lines having GMR_ baked in might be an issue.

**Comment by Robbie1977** [comment]
It is the Individual example Image but that was the example the NBLAST analysis was based on so I think it's more valid.
We could return the Class but then would require extra steps to find an example image to display with target

**Comment by Robbie1977** [comment]
@jefferis how's a fix coming? Hoping to get this live for neurofly.

**Comment by jefferis** [comment]
closed by 8a3b39b

**Comment by Robbie1977** [comment]
getting the occasional error:
```
replacement has 39 rows, data has 40

In call:
`[[<-.data.frame`(`*tmp*`, "id", value = c("VFB_00021833", "VFB_00021915", 
"VFB_00021465", "VFB_00022468", "VFB_00021992", "VFB_00021621", 
"VFB_00021516", "VFB_00021575", "VFB_00022546", "VFB_00022215", 
"VFB_00022228", "VFB_00022249", "VFB_00021175", "VFB_00021216", 
"VFB_00021295", "VFB_00021283", "VFB_00021133", "VFB_00020527", 
"VFB_00020466", "VFB_00020602", "VFB_00020924", "VFB_00020861", 
"VFB_00020841", "VFB_00020880", "VFB_00020719", "VFB_00020734", 
"VFB_00020670", "VFB_00022975", "VFB_00020372", "VFB_00023253", 
"VFB_00023243", "VFB_00023052", "VFB_00023044", "VFB_00023070", 
"VFB_00022614", "VFB_00022663", "VFB_00022807", "VFB_00023142", 
"VFB_00023199"))
```
